src/bptl/work_units/zac/tasks.py

```diff
@@ -0,0 +1,70 @@
+import json
+
+from requests import Response
+
+from bptl.tasks.base import WorkUnit, check_variable
+from bptl.tasks.registry import register
+
+from .client import ZACClient
+from .serializers import ZacUsersDetailsSerializer
+
+
+@register
+class UserDetailsTask(WorkUnit):
+    """
+    Requests email and name data from usernames from the zac.
+
+    In the camunda process models accorderen/adviseren we have a list of usernames from
+    the zac. In order to send signalling emails, we will need to fetch
+    the email addresses and names from the zac and feed it back to the camunda process.
+
+    In this first implementation a simple and direct get request is done
+    at the zac.accounts.api endpoint.
+
+    The task requires the service with alias "zac" to be configured in the work_unit
+    services and be registered in the ZAC where the token will be set.
+
+    **Required process variables**
+
+    * ``usernames``: JSON with usernames.
+        .. code-block:: json
+                [
+                    "user1",
+                    "user2",
+                    "user3",
+                ]
+
+    **Sets the process variables**
+
+    * ``userData``: a JSON-object containing a list of user names and emails:
+
+      .. code-block:: json
+            [
+                {
+                    "name": "FirstName LastName"
+                    "email": "test@test.nl"
+                },
+            ]
+
+    """
+
+    def get_client_response(self) -> Response:
+        client = ZACClient()
+        variables = self.task.get_variables()
+        usernames = check_variable(variables, "usernames")
+        url = f"accounts/api/users?filter_users={','.join(usernames)}&include=True"
+        response = client.get(url)
```

Comment:
I really dislike building querystrings by hand:

- no escaping of inputs happened
- not readable

Please rely on `requests`' abstraction - pass in the query parameters as a dictionary to the underlying `requests.get` call, and let requests handle the querystring/URL building with the appropriate URL quoting.

Also, what happens if some wacky username contains a comma? :thinking: 

The typical approach for multi-dimensional query string arguments is to repeat the argument:

`?username=foo&username=bar&username=baz` - I'm pretty sure [django-filter](https://django-filter.readthedocs.io/en/stable/) can handle that.

Reply:
I suggested that to Kelvin as well but since we're not doing that from the frontend either (check Kelvin's implementation) I'd rather stay consistent...

Reply:
Done

Reply:
> I suggested that to Kelvin as well but since we're not doing that from the frontend either (check Kelvin's implementation) I'd rather stay consistent...

we shall bring the frontend in line with this approach then :-) good "check" to keep in mind when reviewing frontend-code.